### ISSUE TITLE
Problem with Country Map slice:  ['ISO3166'] not in index

### ISSUE THREAD

**Issue by michelepetito** [apache/incubator-superset#3737]
[superset error.zip](https://github.com/apache/incubator-superset/files/1424257/superset.error.zip)

- [x ] I have checked the superset logs for python stacktraces and included it here as text if any
- [ ] I have reproduced the issue with at least the latest released version of superset
- [x ] I have checked the issue tracker for the same issue and I haven't found one similar

### Superset version
0.19.1

### Expected results
The country map. 

### Actual results
```
"['ISO3166'] not in index"
```

```
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/superset/viz.py", line 251, in get_payload
    data = self.get_data(df)
  File "/usr/local/lib/python3.6/site-packages/superset/viz.py", line 1306, in get_data
    ndf = df[cols]
  File "/usr/local/lib/python3.6/site-packages/pandas/core/frame.py", line 1958, in __getitem__
    return self._getitem_array(key)
  File "/usr/local/lib/python3.6/site-packages/pandas/core/frame.py", line 2002, in _getitem_array
    indexer = self.loc._convert_to_indexer(key, axis=1)
  File "/usr/local/lib/python3.6/site-packages/pandas/core/indexing.py", line 1231, in _convert_to_indexer
    raise KeyError('%s not in index' % objarr[mask])
KeyError: "['ISO3166'] not in index"
```

### Steps to reproduce
1) Use "redditi.csv" as table source
2) Create a new coloumn named "ISO3166", type: VARCHAR(10), Groupable, Filterable and with sql expression "trim(concat('IT-', sigla_provincia))".
3) Create a new slice, with visualization type "country map"
4) Use as "ISO 3166-1 codes of region/province/department"  the field "ISO3166" and "Redditi Complessivi" (or "Redditi da fabbricato) for metrics.

**Comment by xrmx** [comment]
Please reproduce with latest release.

**Comment by xrmx** [comment]
Also how are you using a csv as table source?

**Comment by michelepetito** [comment]
I can't reproduce in this moment with last release because I'm using a remote instance of superset that is managed from external administrators (Team per l'Italia Digitale). In the next few days I'll do a local test with last release on my machine.

Yes, I used a csv file, but not directly: I send it to administrators that loaded it into DAF (Data Analytics Framework - https://teamdigitale.governo.it/it/projects/daf.htm -
 https://github.com/italia/daf), an italian govermnet project that use superset as system to create dashboards from italian open data.

**Comment by xrmx** [comment]
FYI the *add column* from the superset ui does not create the column in the db, just tells superset that there's that column on the db.

**Comment by xrmx** [comment]
Does it work for you if you remove the type? On the computed columns i have here i don't have the types.

**Comment by michelepetito** [comment]
I removed the types of columns but it does not work. 

Now I have reproduced the issue with the last release: the country map works!

**Comment by xrmx** [comment]
If with latest release it works then you *haven't* reproduced it :) Please close then.

**Comment by michelepetito** [comment]
Ok, It works also in the release 0.19.1: I created a new column, identical to the previous ISO3166 but I called it "codiceprovincia".

**Comment by xrmx** [comment]
If the db is postgresql then should be fixed by #3736
